### ISSUE TITLE
Add more spam patterns

### ISSUE THREAD

**Issue by tfmorris** [hornc/ol-spam-finder#1]
- Chinese IM Ten Cent (used a lot in transcript spam) - QQ and/or 微信
    https://openlibrary.org/people/kcvnbgj45545
    https://openlibrary.org/search?q=%E5%BE%AE%E4%BF%A1&sort=new

- in-country phone numbers e.g. 0500336791

- variants of .com using funky UTF-8 (these might all be covered already)

。	\xe3\x80\x82
． \xef\xbc\x8e  Full Width Full Stop
｡	\xef\xbd\xa1	HALFWIDTH IDEOGRAPHIC FULL STOP
ｃ	\xef\xbd\x83	FULLWIDTH LATIN SMALL LETTER C
Ｃ	\xef\xbc\xa3	FULLWIDTH LATIN CAPITAL LETTER C
ｏ	\xef\xbd\x8f	FULLWIDTH LATIN SMALL LETTER O
Ｏ	\xef\xbc\xaf	FULLWIDTH LATIN CAPITAL LETTER O
ｍ	\xef\xbd\x8d	FULLWIDTH LATIN SMALL LETTER M
Ｍ	\xef\xbc\xad	FULLWIDTH LATIN CAPITAL LETTER M

- lucky 777 (optionally with whitespace):
  - https://openlibrary.org/search?q=777&sort=new
  - https://openlibrary.org/search?q="7 7 7"

**Comment by hornc** [comment]
Hi @tfmorris, thanks for the tips! I have just pushed a commit that covers some of these that I noticed recently too, most notably all those 777 ones: 73545667c8d48602116dd44e045c772cb4948cda

I'll review to see I have included all those character variations.

I have also added some tests so it's clear what kind of things are caught, and I can quickly check if a new style of spam is already covered.

I had noticed the 微信 giving good results in search, and I have a script combo that will revert all users from a particular search term. I've been using it to revert spam back to 2013.

**Comment by tfmorris** [comment]
Thanks for the quick response! 

Here's one more .com variation: ⓒⓞⓜ

**Comment by tfmorris** [comment]
I haven't had a chance to look at these yet to see why they were missed, but tossing them here so I don't lose them:

https://openlibrary.org/people/st6we4ry64
https://openlibrary.org/people/upkexbt4302
https://openlibrary.org/people/minsu
https://openlibrary.org/people/payne28
https://openlibrary.org/people/filmjavcom
https://openlibrary.org/people/raichel

**Comment by tfmorris** [comment]
We're still getting transcript spam with the Chinese characters for QQ: 微信

e.g.
https://openlibrary.org/people/nokajv09
https://openlibrary.org/people/knaklv02

The second one seems to be a variant with an ampersand between the two characters: 微&信

**Comment by hornc** [comment]
Thanks @tfmorris, those did slip through. The transcript ones seem to be the most persistent spammers, I stopped monitoring them so closely since they were getting picked up, but obviously they are still experimenting with variations to evade the spam checker. I've added a matcher for them in this commit https://github.com/hornc/ol-spam-finder/commit/1f89e0140bf297ba8e83a15137ae1d34a2d0d07c and they have now been removed from the site.

I'll close this issue to keep things focussed as I believe I have covered everything you have listed above, feel free to raise more issues if you notice anything else has slipped through! Cheers!
